apps/balancer-tools/src/app/apr/(utils)/calculatePoolStats.ts

```diff
@@ -206,15 +207,54 @@ export async function calculatePoolStats({
   };
 }
 
-function calculateRoundAPR(
-  round: Round,
+function getWeeksApart(startAt: Date, endAt: Date) {
+  const millisecondsInAWeek = 7 * 24 * 60 * 60 * 1000;
+  const differenceInMilliseconds = Math.abs(
+    startAt.getTime() - endAt.getTime(),
+  );
+  return Math.floor(differenceInMilliseconds / millisecondsInAWeek);
+}
+
+function calculateAPRForDateRange(
+  startAt: Date,
+  endAt: Date,
   votingShare: number,
   tvl: number,
   balPriceUSD: number,
   feeAPR: number,
   tokensAPR: tokenAPR[],
 ) {
-  const emissions = balEmissions.weekly(round.endDate.getTime() / 1000);
+  const weeksApart = getWeeksApart(startAt, endAt);
+  let emissions;
+  if (weeksApart >= 1) {
+    const weekArray = Array.from({ length: weeksApart }, (_, index) => {
+      const weekStartDate = new Date(startAt);
+      weekStartDate.setDate(startAt.getDate() + index * 7);
+      const weekEndDate = new Date(weekStartDate);
+      weekEndDate.setDate(weekStartDate.getDate() + 6);
+      return { weekNumber: index + 1, weekStartDate, weekEndDate };
+    });
+
+    // Calculate the total balance emissions and count of weeks
+    const { totalBalanceEmissions, weekCount } = weekArray.reduce(
+      (acc, week) => {
+        const weeklyBalanceEmissions = balEmissions.weekly(
+          week.weekStartDate.getTime() / 1000,
+        );
+        return {
+          totalBalanceEmissions:
+            acc.totalBalanceEmissions + weeklyBalanceEmissions,
+          weekCount: acc.weekCount + 1,
+        };
+      },
+      { totalBalanceEmissions: 0, weekCount: 0 },
+    );
+
+    emissions = totalBalanceEmissions / weekCount;
```

Comment:
the emissions might not be the same for every week, they'll change every change of year. So if the period is between the yearly cliff this will return an average, instead of a cliff on the graph, which is not what should appear there. makes sense? we can split another task to rework this though

Reply:
https://linear.app/bleu-llc/issue/BAL-876/fix-totalbalanceemissions-calculations-when-fetching-on-year-change